moto/iam/config.py

```diff
@@ -0,0 +1,221 @@
+import json
+
+from moto.core.exceptions import InvalidNextTokenException
+from moto.core.models import ConfigQueryModel
+from moto.iam import iam_backends
+
+CONFIG_BACKEND_DELIM = "\x1e"  # Record Seperator "RS" ASCII Character
+
+
+class RoleConfigQuery(ConfigQueryModel):
+    def list_config_service_resources(
+        self,
+        resource_ids,
+        resource_name,
+        limit,
+        next_token,
+        backend_region=None,
+        resource_region=None,
+    ):
+        # IAM roles are "global" and aren't assigned into any availability zone
+        # The resource ID is a AWS-assigned random string like "AROA0BSVNSZKXVHS00SBJ"
+        # The resource name is a user-assigned string like "MyDevelopmentAdminRole"
+        # Stored in moto backend with the AWS-assigned random string like "AROA0BSVNSZKXVHS00SBJ"
+
+        # Grab roles from backend; need the full values since names and id's are different
+        role_list = list(self.backends["global"].roles.values())
+
+        if not role_list:
+            return [], None
+
+        # Filter by resource name or ids
```

Comment:
Typo on precedence.

Also, for aggregation you need to add a check in for both resource names and resource IDs provided. You will need do something along the lines of:
```
        if resource_name or resource_ids:
            filtered_roles = []
            # resource_name takes precedence over resource_ids
            if resource_name:
                for role in role_list:
                    if role.name == resource_name:
                        filtered_roles = [role]
                        break

                # For aggregation: if both are passed in, need to verify both values are correct.
                if resource_ids:   
                    if role.id not in resource_ids:
                        return [], None

            else:
                for role in role_list:
                    if role.id in resource_ids:
                        filtered_roles.append(role)
```